lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1639,13 +1639,56 @@ SPIRVValue *LLVMToSPIRV::oclTransSpvcCastSampler(CallInst *CI,
   return BV;
 }
 
-std::vector<std::pair<Decoration, std::string>>
-tryParseIntelFPGAAnnotationString(StringRef AnnotatedCode) {
-  std::vector<std::pair<Decoration, std::string>> Decorates;
+using DecorationsInfoVec = std::vector<std::pair<Decoration, std::string>>;
+
+struct IntelLSUControlsInfo {
+  void setWithBitMask(unsigned ParamsBitMask) {
+    if (ParamsBitMask & 0x1)
+      BurstCoalesce = true;
+    if (ParamsBitMask & 0x2)
+      CacheSizeInfo = 0;
+    if (ParamsBitMask & 0x4)
+      DontStaticallyCoalesce = true;
+    if (ParamsBitMask & 0x8)
+      PrefetchInfo = 0;
+  }
+
+  DecorationsInfoVec getDecorationsFromCurrentState() {
+    DecorationsInfoVec ResultVec;
+    // Simple flags
+    if (BurstCoalesce)
+      ResultVec.emplace_back(DecorationBurstCoalesceINTEL, "");
+    if (DontStaticallyCoalesce)
+      ResultVec.emplace_back(DecorationDontStaticallyCoalesceINTEL, "");
+    // Conditional values
+    int CacheSize = 0;
+    if (CacheSizeInfo.hasValue()) {
+      CacheSize = CacheSizeInfo.getValue();
+      auto DecorationString = std::to_string(CacheSize);
+      ResultVec.emplace_back(DecorationCacheSizeINTEL, DecorationString);
+    }
+    int PrefetchSize = 0;
+    if (PrefetchInfo.hasValue()) {
+      PrefetchSize = PrefetchInfo.getValue();
+      ResultVec.emplace_back(DecorationPrefetchINTEL,
```

Comment:
I like this version more, then one for `CacheSize` from above. But what about making it be a one-liner? (getting rid of temp variables).

Reply:
Sure. An even better approach could be to have the prefetch/cache-size's `Optional`s hold a `StringRef`, thus avoiding the string -> integer -> string conversions. What do you think here?

Reply:
I like it

Reply:
I've decided to go with your initial suggestion so that the intermediate "integer step" can be used as a value verification at caller site.